lib/common/parse-features-string.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * Utilities to parse comma-separated key value pairs used in browser APIs.
+ * For example: "x=100,y=200,width=500,height=500"
+ */
+import { BrowserWindowConstructorOptions } from 'electron';
+
+type RequiredBrowserWindowConstructorOptions = Required<BrowserWindowConstructorOptions>;
+type IntegerBrowserWindowOptionKeys = {
+  [K in keyof RequiredBrowserWindowConstructorOptions]:
+    RequiredBrowserWindowConstructorOptions[K] extends number ? K : never
+}[keyof RequiredBrowserWindowConstructorOptions];
+
+// This could be an array of keys, but an object allows us to add a compile-time
+// check validating that we haven't added an integer property to
+// BrowserWindowConstructorOptions that this module doesn't know about.
+const keysOfTypeNumberCompileTimeCheck: { [K in IntegerBrowserWindowOptionKeys] : true } = {
+  x: true,
+  y: true,
+  width: true,
+  height: true,
+  minWidth: true,
+  maxWidth: true,
+  minHeight: true,
+  maxHeight: true,
+  opacity: true
+};
+// Note `top` / `left` are special cases from the browser which we later convert
+// to y / x.
+const keysOfTypeNumber = ['top', 'left', ...Object.keys(keysOfTypeNumberCompileTimeCheck)];
+
+/**
+ * Note that we only allow "0" and "1" boolean conversion when the type is known
+ * not to be an integer.
+ *
+ * The coercion of yes/no/1/0 represents best effort accordance with the spec:
+ * https://html.spec.whatwg.org/multipage/window-object.html#concept-window-open-features-parse-boolean
+ */
+type CoercedValue = string | number | boolean;
+function coerce (key: string, value: string): CoercedValue {
+  if (keysOfTypeNumber.includes(key)) {
+    return Number(value);
+  }
+
+  switch (value) {
+    case 'true':
+    case '1':
+    case 'yes':
+    case undefined:
+      return true;
+    case 'false':
+    case '0':
+    case 'no':
+      return false;
+    default:
+      return value;
+  }
+}
+
+function parseCommaSeparatedKeyValue (source: string, useSoonToBeDeprecatedBehaviorForBareKeys: boolean) {
+  const bareKeys = [] as string[];
+  const parsed = source.split(',').reduce((map, keyValuePair) => {
+    const [key, value] = keyValuePair.split('=').map(str => str.trim());
+
+    if (useSoonToBeDeprecatedBehaviorForBareKeys && value === undefined) {
+      bareKeys.push(key);
+      return map;
+    }
+
+    map[key] = coerce(key, value);
+    return map;
+  }, {} as { [key: string]: any });
```

Comment:
i'm a big fan of `reduce` but i think a `forEach` or a loop would be easier to read here.

```suggestion
  const parsed = {} as { [key: string]: any };
  for (const keyValuePair of source.split(',')) {
    const [key, value] = keyValuePair.split('=').map(str => str.trim());

    if (useSoonToBeDeprecatedBehaviorForBareKeys && value === undefined) {
      bareKeys.push(key);
      continue;
    }

    parsed[key] = coerce(key, value);
  }
```

Reply:
I'm into it.